include/domain/impl/receivers/kernel.tpp

```diff
@@ -186,10 +194,110 @@ void specfem::domain::impl::kernels::receiver_kernel<
+              if(seismogram_type_l == specfem::enums::seismogram::type::pressure){
+                //for pressure, we need to compute -kappa * div(s)
+                //so we store displacement into auxfield so we can overwrite
+                //sv_receiver_field
+
+                #ifndef KOKKOS_ENABLE_CUDA
+                #pragma unroll
+                #endif
+                for (int l = 0; l < specfem::dimension::dimension<DimensionType>::dim; l++) {
+                  aux_field.displacement(iz,ix,l) = sv_receiver_field(l);
+                }
+              }
             });
-
         team_member.team_barrier();
 
+        if(seismogram_type_l == specfem::enums::seismogram::type::pressure){
+          // we stored displacement into auxfield, and we need to compute -kappa * div(s)
+          // algorithms::divergence might work, but it seems to take a chunk_policy,
+          // which we don't have right now.
+
+          Kokkos::parallel_for(
+            quadrature_points.template TeamThreadRange<specfem::enums::axes::z,
+                                                       specfem::enums::axes::x>(
+                team_member),
+            [=](const int xz) {
+              int iz, ix;
+              sub2ind(xz, ngllx, iz, ix);
+              const specfem::point::index<DimensionType> index(ispec_l, iz, ix);
+              const auto point_partial_derivatives =
+                  [&]() {
+                    specfem::point::partial_derivatives<DimensionType, false,
+                                                        using_simd>
+                    point_partial_derivatives;
+                specfem::compute::load_on_device(index, partial_derivatives,
+                                                 point_partial_derivatives);
+                return point_partial_derivatives;
+              }();
+
+              const auto point_properties =
+                  [&]() -> specfem::point::properties<DimensionType, MediumTag, PropertyTag, using_simd> {
+                specfem::point::properties<DimensionType, MediumTag, PropertyTag, using_simd>
+                    point_properties;
+                specfem::compute::load_on_device(index, properties,
+                                                 point_properties);
+                return point_properties;
+              }();
+
+
+              //bulk modulus
+              const type_real kappa = [&]() -> type_real {
+                if constexpr (MediumTag == specfem::element::medium_tag::acoustic){
+                  return point_properties.kappa;
+                }else if constexpr (MediumTag == specfem::element::medium_tag::elastic){
+                  //we may also need an isotropic if statement?
+                  return point_properties.lambdaplus2mu;
+                }else{
+                  static_assert(false, "include/domain/impl/receivers/kernel.tpp: unknown medium to retrieve kappa!");
+                  return 0;
+                }
+              }();
```

Comment:
<span id="comment-bulk-modulus-params-code"></span>These may or may not be the right properties for the bulk modulus. Please verify.

Reply:
([link to return to checklist](#desc-devlist-checkbox-bulk_modulus))